.NET/Microsoft.Recognizer.Text.TimeZone/English/TimeZoneParser.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using Microsoft.Recognizers.Definitions.English;
+
+namespace Microsoft.Recognizers.Text.TimeZone.English
+{
+    class TimeZoneParser : IParser
+    {
+        public TimeZoneParser()
+        {
+
+        }
+        public int getMinutes(string matched)
+        {
+            int bs = 1;
+            if (matched.StartsWith("+")
+                || matched.StartsWith("-"))
+            {
+                if (matched.StartsWith("-"))
+                    bs = -1;
+
+                matched = matched.Substring(1).Trim();
+            }
+            int h = 0;
+            int m = 0;
+            if (matched.Contains(":"))
+            {
+                string f1 = matched.Split(':')[0];
+                string f2 = matched.Split(':')[1];
+                h = int.Parse(f1);
+                m = int.Parse(f2);
+            }
+            else if (int.TryParse(matched, out h))
+            {
+                m = 0;
+            }
+
+            if (h > 12)
+            {
+                return -10000;
+            }
+
+            if (m != 0 && m != 15 && m != 30 && m != 45 && m != 60)
+            {
+                return -10000;
+            }
+            int totalm = h * 60 + m;
+            totalm *= bs;
+            return totalm;
+        }
+        public ParseResult Parse(ExtractResult extResult)
+        {
+            ParseResult ret = null;
+            string extra;
+            ret = new ParseResult
+            {
+                Start = extResult.Start,
+                Length = extResult.Length,
+                Text = extResult.Text,
+                Type = extResult.Type
+            };
+            if ((extra = extResult.Data as string).Equals("DirectUTC"))
+            {
+                string text = extResult.Text.ToLower();
+                string matched = Regex.Match(text, TimeZoneDefinitions.DirectUTCRegex).Groups[2].Value;
+                int tmpMinutes = getMinutes(matched);
+                if(tmpMinutes!=-10000)
+                {
+                    ret.Value = tmpMinutes;
+                    ret.ResolutionStr = "UTC MINUTES SHIFT: " + tmpMinutes.ToString();
```

Comment:
If "UTC MINUTES SHIFT: " is a special string expected in output or further parsing, please add a constant for it.
It makes it much easier to avoid mistakes and to change in the code in the future.